### DISCUSSION TITLE
Have anyone successfully booted VM instance on top of RK3588?

### DISCUSSION THREAD

**Discussion by Ruach** [Joshua-Riek/ubuntu-rockchip#856] [Q&A]
Hi all, I would like to run VM on orangepi-5-plus with the ubuntu-rockchip.
I could successfully patched the KVM source code of the ubuntu-rockchip and booted up the board.
Instead of compiling another kernel to boot up linux VM, I would like to utilize the existing host image, located in /boot.

This is the command what I've used for running VM 

> lkvm run -i /boot/initrd.img -k /boot/vmlinuz -m 1024 -c 8 --disk= rootfs.img

I passed the initrd.img as an initrd and pass the installed kernel image for booting up the guest.
It seems that kernel has been successfully started, but at some point, it jumps into the busybox after running /init

<img width="1266" alt="Screen Shot 2024-06-01 at 4 48 26 PM" src="https://github.com/Joshua-Riek/ubuntu-rockchip/assets/10967239/0a86a311-f49c-46e1-90b6-1a29436d1a36">
 
Based on error message seems that dev/vda cannot be located 
Also, when I run $ls /dev
it doesn't show any proper devices for booting up rootfs.

It might be not a proper place to ask this question, but I was wondering whether this is because the host image used for booting guest VM is compiled for orangepi from SD-CARD. 

If anyone has successfully ran the guest VM on top of rk3588 please educate me

**Comment by Joshua-Riek** [comment]
see https://github.com/Joshua-Riek/ubuntu-rockchip/issues/731

**Reply by Ruach** [reply]
Thanks a lot, but it seems that the #731 talks about issues when rk3588 runs qemu. I've used lkvm (kvmtool) and could successfully boot the kernel but having a problem because it keeps fall back to initramfs...

**Reply by Ruach** [reply]
@Joshua-Riek  I was wondering if it's because the kernel has been compiled to be loaded from MMC?

**Reply by Joshua-Riek** [reply]
If your dropping to the initrd id check the kernel command line, i'm not sure what the contents of rootfs.img are, but its likely not pointing to the rootfs partition.

**Reply by Ruach** [reply]
@Joshua-Riek  Thanks, when I check the kernel command line in initramfs on guest VM, it is set as /dev/vda. 
Also I checked the content of initrd.img of the host, it doesn't contain any other directories but the usr/lib/firmware...
I am not sure how the linux kernel run /init even though the initrd has only usb/lib/firmware..

**Reply by Ruach** [reply]
@Joshua-Riek  I tried to exclude the vendor specific drivers to initialize the storage such as MMC, but still the guest VM cannot be booted.. hmm it's really weird and I don't know why it cannot locate any disk under /dev/

**Reply by Joshua-Riek** [reply]
if you provide exact steps that can be reproduced I can look at the drivers.
